### PR TITLE
Add a CLI app

### PR DESCRIPTION
**Pull request by TimothyGu**
This is a WIP, DO NOT MERGE. It is already usable, but I need some time to address the to-dos below.

Ported from Jade.

To-do:

1. Documentation
2. Tests
3. Changelog

### CODE REVIEW

**Comment by mde** [comment]
This is good stuff. I'd like to see a -v/V flag for "version." Also, I'm not a huge fan of "locals." I prefer "data," but I can see how it could be good to keep a similar API with the Jade one. If that is the case, we might want to consider changing the EJS docs to use the same terminology.

There is also some prior art with Erubis: http://www.kuwata-lab.com/erubis/users-guide.07.html#command

**Comment by TimothyGu** [comment]
@mde said:

> This is good stuff. I'd like to see a -v/V flag for "version."

commander handles that automatically. Try pulling it and do `bin/ejs.js -V`.

File: bin/ejs.js [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+  , dirname = path.dirname
+  , resolve = path.resolve
+  , join = path.join
+  , mkdirp = require('mkdirp')
+  , ejs = require('../')
+  , options = {}
+  , locals = {};
+
+program
+  .version(require('../package.json').version)
```
Comment (TimothyGu) [inline bin/ejs.js]: See?

**Comment by mde** [comment]
Ah, nice. It doesn't appear in the usage output, so I assumed it wasn't implemented. Let's add it to the output.

**Comment by mde** [comment]
Ah, I see it -- sorry, I'm not familiar with commander's API. I was just looking at output strings.

**Comment by TimothyGu** [comment]
@mde said:

> Also, I'm not a huge fan of "locals." I prefer "data," but I can see how it could be good to keep a similar API with the Jade one.

... and pretty much the remainder of the node.js world, including express.js. I feel it is more important to be more consistent with the community of the language in which the program is written, than with some far relative written in Ruby. (I don't know any Ruby so just my 2¢.)

> If that is the case, we might want to consider changing the EJS docs to use the same terminology.

I will when I get around to doing that :wink:

**Comment by mde** [comment]
:+1:

**Comment by TimothyGu** [comment]
I'll work on this a little more.

**Comment by mde** [comment]
:+1:

**Comment by raphaelokon** [comment]
Hi. A CLI option would be awesome! Is this branch stale?

**Comment by mde** [comment]
Maybe a little less-than-fresh. If someone wants to pick up the baton, I'd love to land this.

**Pull request closed without merging**